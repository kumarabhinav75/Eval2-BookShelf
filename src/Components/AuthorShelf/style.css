 .BooksContainer {
  display: inline-table;
  width: 300px;
  min-width: 250px;
  margin: 20px;
  min-height: 200px;
}
