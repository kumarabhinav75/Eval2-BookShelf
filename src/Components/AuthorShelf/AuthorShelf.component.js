import React from 'react';
import './style.css';
import BookCard from '../BookCard/BookCard.component';

const AuthorShelf = (props) => {
  // console.log('author',props.books);
  // if(props.books){
  //   props.books.map(book => {
  //     console.log(book.Name);
  //   })
  //  } 
if(props.books){
  return(
    props.books.map((book) =>
    <div className="BooksContainer">
      <BookCard bookData = {book} />
    </div>
    )
  )
  }
else return ' ';
}

export default AuthorShelf;