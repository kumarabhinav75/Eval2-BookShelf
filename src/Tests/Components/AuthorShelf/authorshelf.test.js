import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import AuthorShelf from '../../../Components/AuthorShelf/AuthorShelf.component';

describe('AuthorShelf ', () => {
  it ('Should match the snapshot', () => {
    const tree = renderer.create(<AuthorShelf />);
    expect(tree).toMatchSnapshot();
  });
  it('Should get the props provided', () => {
    const bookData = [{
      Name: 'BookName',
      rating: '4.56'
    }];
    const wrapper = shallow(<AuthorShelf books={bookData} />);
    expect(wrapper.props().children.props.bookData.Name).toEqual(bookData[0].Name);
    expect(wrapper.props().children.props.bookData.rating).toEqual(bookData[0].rating);
  })
})