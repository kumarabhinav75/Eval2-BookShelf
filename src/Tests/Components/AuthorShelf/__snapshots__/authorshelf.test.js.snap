// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorShelf  Should match the snapshot 1`] = `" "`;
